test/screens/play/game_end_page_test.dart

```diff
@@ -0,0 +1,76 @@
+import 'package:flutter_test/flutter_test.dart';
+import 'package:treasure_of_the_high_seas/model/game_result.dart';
+import 'package:treasure_of_the_high_seas/model/game_state_factory.dart';
+import 'package:treasure_of_the_high_seas/screens/main_menu/main_menu_page.dart';
+import 'package:treasure_of_the_high_seas/screens/play/game_end_page.dart';
+import 'package:treasure_of_the_high_seas/screens/play/play_page.dart';
+import 'package:treasure_of_the_high_seas/screens/play/player_hand.dart';
+
+import '../../test_utils.dart';
+
+void main() {
+  testWidgets('should display correct text for win and loss', (WidgetTester tester) async {
+    await tester.pumpWidget(createWidgetForTesting(child: GameEndPage(GameResult.WIN)));
+    expect(find.byType(GameEndPage), findsOneWidget);
+    expect(find.byType(PlayerHand), findsNothing);
+
+    var gameWinTextFinder = find.text(GameResult.WIN.description);
+    var gameLoseTextFinder = find.text(GameResult.LOSE.description);
```

Comment:
Could declare as `final` since they're not being reassigned - I think there's a linter rule you can enable for unnecessary `var` statements.

Might be worth us looking in general at whether there's a static code analysis tool we could run as part of the pipeline to catch things like this, I'll add an issue for it if we don't already have one! 👍

Reply:
I'm confused, they are being reassigned about 5 lines further down. I agree with having more automated stuff though.

Reply:
Ah yeah, didn't notice that. Don't think we need to reassign them at all, actually - if I change them to `final` and remove the reassignments the test still passes for me :+1: